streams/shared/src/main/scala/zio/stream/internal/ZInputStream.scala

```diff
@@ -0,0 +1,95 @@
+package zio.stream.internal
+
+import scala.annotation.tailrec
+
+import zio.Runtime
+import zio.{ Chunk, Exit, FiberFailure, ZIO }
+
+private[zio] class ZInputStream(chunks: Iterator[Chunk[Byte]]) extends java.io.InputStream {
+  private var current: Iterator[Byte]      = Iterator.empty
```

Comment:
Can we get rid of this and keep the current chunk and offsets in a var? This will box the bytes.